### ISSUE TITLE
pulumi new prompts for a passphrase twice when logged into the local backend

### ISSUE THREAD

**Issue by justinvp** [pulumi/pulumi#3077]
```
$ pulumi new typescript
This command will walk you through creating a new Pulumi project.

Enter a value or leave blank to accept the (default), and press <ENTER>.
Press ^C at any time to quit.

project name: (myproj) 
project description: (A minimal TypeScript Pulumi program) 
Created project 'myproj'

Please enter your desired stack name.
To create a stack in an organization, use the format <org-name>/<stack-name> (e.g. `acmecorp/dev`).
stack name: (dev) 
Enter your passphrase to protect config/secrets: 
Re-enter your passphrase to confirm: 
Created stack 'dev'

Enter your passphrase to unlock config/secrets
    (set PULUMI_CONFIG_PASSPHRASE to remember): 
Installing dependencies...
```

Since the passphrase was just entered after choosing a stack name, why prompt for it again before installing dependencies? It should just reuse the passphrase that was just entered.

Related: Why do we proactively prompt for a passphrase when creating the stack? Couldn't this be deferred until the first time config secrets are used? What if I never use any secrets with my stack?

**Comment by ellismg** [comment]
> Related: Why do we proactively prompt for a passphrase when creating the stack? Couldn't this be deferred until the first time config secrets are used? What if I never use any secrets with my stack

We could defer asking this until you do your initial update or do `pulumi config set --secret`, but we will need to ask it before an initial update for the reasons outlined in https://github.com/pulumi/pulumi/issues/2728

**Comment by justinvp** [comment]
This was fixed by #8986
